scalafix/rules/src/main/scala/fix/ChiselLinter.scala

```diff
@@ -0,0 +1,41 @@
+package fix
+
+import scalafix.v1._
+import scala.meta._
+
+case class AmbiguousLiteralExtract(conv: String, valLit: Lit.Int, bitLit: Lit.Int) extends Diagnostic {
+  override def position: Position = valLit.pos
+  override def message: String =
+    s"Passing an integer to .${conv} does not set width but does a bit extract. If you really want this bit extract, use ${valLit}.${conv}()(${bitLit})"
```

Comment:
This suggestion depends on adding parens to the definitions of `.U` et al. I know @jackkoenig was considering doing that, since it makes it easier to protect against the immediate-bit-extract construct.